docs: add Walrus Console beta documentation (#3540)
MystenLabs/walrus: docs/content/console/storage-epochs.mdx

---
title: Storage, Epochs, and Renewal
description: How Walrus Console measures storage in epochs, how automatic renewal keeps active accounts from losing data, and how dormant accounts expire.
keywords: [walrus console, storage, epochs, renewal, auto-renewal, expiry]
---

Walrus storage is time-bound. Your data stays available for a set amount of storage, measured in epochs, and expires when that storage runs out. Walrus Console tracks and pays for this so you do not manage epochs by hand, and it is built to renew storage automatically for accounts that stay active.

:::info

> Storage and epochs work today. Automatic renewal ships at and after GA. This page explains both so you understand how your data's lifetime works and how to avoid losing it. For related cost details, see the [concepts and overview](./overview).

Reviewer: > Automatic renewal ships at and after GA. 

We're having automatic renewal and on Beta

:::

## How storage epochs work

Walrus measures storage in epochs, which are fixed periods defined by the Walrus network. When you store a file, you reserve storage for a number of epochs. While that storage is funded, the network keeps your data available. When it runs out and you do not renew it, the data expires and is no longer retrievable.

Console handles the underlying payment and epoch accounting for you through your [Pearl wallet](./auth#your-pearl-wallet). You store a file, and Console reserves and funds its storage. You do not buy epochs or sign renewal transactions yourself.

## Automatic renewal

To keep active developers from losing data because they forgot to renew, Console renews storage automatically for wallets that stayed active. A wallet counts as active when it has at least one Walrus transaction within a recent activity window, planned as 14 days. Renewal runs on your behalf, so data you keep using stays available without any manual step.

> Automatic renewal ships at and after GA, alongside Mainnet billing.

Reviewer: Beta will also support automatic renewals

## Dormant accounts and expiry

:::warning

If a wallet goes dormant, its storage lapses and its data expires. Automatic renewal covers active wallets only. If you store data you want to keep, make sure the account stays active within the activity window, or plan to re-upload.

:::

Expiring dormant storage is deliberate. It keeps the network from paying indefinitely to store data for accounts that are no longer in use. An account returns to active status as soon as it records a new Walrus transaction within the window.

## Checking storage status

> Your dashboard shows how much storage each space is using and surfaces upcoming expiry so you can act before data lapses. Storage used and your storage cap also appear per space when you list spaces through the API.

Reviewer: > surfaces upcoming expiry so you can act before data lapses.

We don't have this

---

> title: Storage, epochs, and renewal

Reviewer 2: ```suggestion
title: Storage, Epochs, and Renewal
```

description: How Walrus Console measures storage in epochs, how automatic renewal keeps active accounts from losing data, and how dormant accounts expire.
keywords: [walrus console, storage, epochs, renewal, auto-renewal, expiry]
---

Walrus storage is time-bound. Your data stays available for a set amount of storage, measured
in epochs, and expires when that storage runs out. Walrus Console tracks and pays for this so
you do not manage epochs by hand, and it is built to renew storage automatically for accounts
that stay active.

> :::note

Reviewer 2: ```suggestion
:::info
```

> Storage and epochs work today. Automatic renewal ships at and after GA. This page explains both

Reviewer 2: ```suggestion
Automatic renewal will ship at and after GA. This page explains both
```

Reviewer 2: again, need some consistency on when these docs plan to be out and if it should include alpha explicit info or skip to beta

so you understand how your data's lifetime works and how to avoid losing it. For related cost
details, see the [concepts and overview](./overview).

:::

## How storage epochs work

Walrus measures storage in epochs, which are fixed periods defined by the Walrus network. When
you store a file, you reserve storage for a number of epochs. While that storage is funded, the
network keeps your data available. When it runs out and is not renewed, the data expires and is
no longer retrievable.

> Console handles the underlying payment and epoch accounting for you through your Pearl wallet.

Reviewer 2: link to pearl wallet definition

renewal covers active wallets only. If you store data you want to keep, make sure the account
stays active within the activity window, or plan to re-upload.

:::

Expiring dormant storage is deliberate. It keeps the network from paying indefinitely to store
data for accounts that are no longer in use. An account returns to active status as soon as it
records a new Walrus transaction within the window.

## Checking storage status

Your dashboard shows how much storage each space is using and surfaces upcoming expiry so you
can act before data lapses. Storage used and your storage cap also appear per space when you
list spaces through the API.

## Storage and the free tier

Console keeps a perpetual free tier with a storage cap, tentatively 5 GB, so new developers are
not paywalled. Usage-based billing for reads and egress, along with a paid path, follows at and
after GA. Console manages WAL token handling for storage on your behalf. For the model behind
spaces and storage, see the [concepts and overview](./overview).

## Next steps

- Follow the [quickstart](./quickstart) to store your first file.

> - See the [concepts and overview](./overview) for spaces, buckets, and asset types.

Reviewer 2: ```suggestion

```
